### PR TITLE
API Usage Monitoring: Client Metrics

### PR DESCRIPTION
**Pull request by ddufourboivin-zalando**
A new set of metrics is now available for tracking client-oriented (aka caller, consumer) information. They -- along with the new metrics format -- are documented in filter.md in the documentation (part of this pull request).

### CODE REVIEW

File: filters/apiusagemonitoring/filter.go [hunk 1]
```diff
@@ -70,49 +154,93 @@ func (f *apiUsageMonitoringFilter) String() string {
...
+func getEndpointMetricsNames(req *http.Request, path *pathInfo) *endpointMetricNames {
 	method := req.Method
 	methodIndex, ok := methodToIndex[method]
 	if !ok {
-		methodIndex = MethodIndexUnknown
+		methodIndex = methodIndexUnknown
 		method = unknownElementPlaceholder
 	}
 
 	prefixes := path.metricPrefixesPerMethod[methodIndex]
```
Comment (szuecs) [inline filters/apiusagemonitoring/filter.go]: I would rather do it similar as above:

```go
prefixes, ok := path.metricPrefixesPerMethod[methodIndex]
if ok { return prefixes }
return createAndCacheMetricsNames(path, method, methodIndex)
```

Comment (ddufourboivin-zalando) [reply filters/apiusagemonitoring/filter.go]: This is an `array`, not a `map`. There is no `ok` 2nd return value.

Comment (ddufourboivin-zalando) [reply filters/apiusagemonitoring/filter.go]: I could do something like 

```go
	if p := path.metricPrefixesPerMethod[methodIndex]; p != nil {
		return p
	}
	return createAndCacheMetricsNames(path, method, methodIndex)
```

to be more go-ish.

**Comment by ddufourboivin-zalando** [comment]
@szuecs : converted to backticks in 0a660cbb446afa66271b4dafe705df023b8bedaf

**Comment by ddufourboivin-zalando** [comment]
👍

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
>  I'd like to simply confirm / establish with you that I was this "ghost" who opened it in the first place.

Hi, there is you name in the commit history
Screenshot from 2023-12-05 17-48-08

```
$ git show 155ce16 | head
commit 155ce162ac1801f0e541161b3ba88815f0b5b3c2
Author: David Dufour-Boivin <39987010+ddufourboivin-zalando@users.noreply.github.com>
Date:   Wed Nov 14 16:15:32 2018 +0100

    API Usage Monitoring: Client Metrics (#844)
    
    * client based metrics
    
    * always implement Stringer on non-pointer receiver
```

**Comment by Adeynack** [comment]
@AlexanderYastrebov : Of course 🤦🏼 the commits are signed and named! Thanks :-)

**Pull request merged**